# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/xpconnect/src/XPCJSRuntime.cpp
# Commit: a3fd24f5f0a7
# Full Hash: a3fd24f5f0a7bc094409fdbdbe1ef92fc459677b
# Author: Jan de Mooij <jdemooij@mozilla.com>
# Date: 2019-11-23 21:49:00
# Regressor Bug: 1579367
# File Overlap Count: 1
# Description:
#   Bug 1579367 - Initialize XPCJSContext explicitly, after loading user prefs. r=kmag
#   
#   This way we get the correct values for start-up prefs in the parent process.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D51061
# ==============================================================================

diff -r e06f2a4de969 -r a3fd24f5f0a7 js/xpconnect/src/XPCJSRuntime.cpp
--- a/js/xpconnect/src/XPCJSRuntime.cpp	Fri Nov 22 09:48:58 2019 +0000
+++ b/js/xpconnect/src/XPCJSRuntime.cpp	Fri Nov 22 12:40:17 2019 +0000
@@ -32,6 +32,7 @@
 #include "nsIPlatformInfo.h"
 #include "nsPIDOMWindow.h"
 #include "nsPrintfCString.h"
+#include "nsScriptSecurityManager.h"
 #include "nsThreadPool.h"
 #include "nsWindowSizes.h"
 #include "mozilla/Preferences.h"
@@ -1120,6 +1121,8 @@
 
   JS::SetGCSliceCallback(cx, mPrevGCSliceCallback);
 
+  nsScriptSecurityManager::GetScriptSecurityManager()->ClearJSCallbacks(cx);
+
   // Shut down the helper threads
   gHelperThreads->Shutdown();
   gHelperThreads = nullptr;
@@ -3050,6 +3053,8 @@
   // these jsids filled in later when we have a JSContext to work with.
   mStrIDs[0] = JSID_VOID;
 
+  nsScriptSecurityManager::GetScriptSecurityManager()->InitJSCallbacks(cx);
+
   // Unconstrain the runtime's threshold on nominal heap size, to avoid
   // triggering GC too often if operating continuously near an arbitrary
   // finite threshold (0xffffffff is infinity for uint32_t parameters).